server-spi-private/src/main/java/org/keycloak/authorization/AdminPermissionsSchema.java

```diff
@@ -183,7 +202,15 @@ public Scope getScope(KeycloakSession session, ResourceServer resourceServer, St
     }
 
     public void init(KeycloakSession session, RealmModel realm) {
-        ClientModel client = session.clients().addClient(realm, Constants.ADMIN_PERMISSIONS_CLIENT_ID);
+        ClientProvider clients = session.clients();
+        ClientModel client = clients.getClientByClientId(realm, Constants.ADMIN_PERMISSIONS_CLIENT_ID);
```

Comment:
just a nitpick, IMO it improve readability.
```suggestion
        ClientModel client = realm.getAdminPermissionsClient();
```

Reply:
I'm updating it.